### PR TITLE
refactor: Move CNI bridge/transparent routes to common

### PR DESCRIPTION
**Pull request by matmerr**
**Reason for Change**:

Fixed an issue for adding routes to the external interface, in transparent and bridge mode, as well as the snat rules

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests

**Notes**:

### CODE REVIEW

File: network/network_linux.go [hunk 1]
```diff
@@ -480,6 +497,84 @@ func (nm *networkManager) disconnectExternalInterface(extIf *externalInterface,
...
+			}
+			log.Printf("[cni-net] route already exists: dst %+v, gw %+v, interfaceName %v", nlRoute.Dst, nlRoute.Gw, route.DevName)
+		}
+
+		log.Printf("[cni-net] Added route %+v", route)
+	}
+
+	return nil
+
+	// old
```
Comment (matmerr) [inline network/network_linux.go:539]: remove commented code

File: iptables/iptables.go [hunk 2]
```diff
@@ -85,8 +85,13 @@ var (
 	DisableIPTableLock bool
 )
 
+type IpTableEntry struct {
```
Comment (matmerr) [inline iptables/iptables.go:88]: nit: `IPTableEntry`

File: cni/network/invoker_cns.go [hunk 3]
```diff
@@ -60,77 +62,102 @@ func (invoker *CNSIPAMInvoker) Add(nwCfg *cni.NetworkConfig, subnetPrefix *net.I
...
+	// this route is needed when the vm on subnet A needs to send traffic to a pod in subnet B on a different vm
+	options[network.RoutesKey] = []network.RouteInfo{
+		{
+			Dst: *ncSubnetPrefix,
+			Gw:  hostGateway,
+		},
+	}
+
+	azureDNSMatch := fmt.Sprintf(" -m addrtype ! --dst-type local -s %s -d %s -p %s --dport %d", ncSubnetPrefix.String(), iptables.AzureDNS, iptables.UDP, iptables.DNSPort)
+	snatPrimaryIPJump := fmt.Sprintf("%s --to %s", iptables.Snat, info.podIPAddress)
```
Comment (matmerr) [inline cni/network/invoker_cns.go:153]: info.podIPAddress don't snat with podIPAddress, snat with ncPrimaryIPAddress

**Review by vakalapa — APPROVED** [review]

**Pull request merged**